### PR TITLE
Entry improvements

### PR DESCRIPTION
**Pull request by PiotrKozimor**
This one is more readable :)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -73,3 +74,7 @@ complement/
 docs/_site
 
 media_store/
+
+__debug_bin
+
+cmd/dendrite-monolith-server/dendrite-monolith-server
```
Comment (Danieloni1) [inline .gitignore:80]: Do we really want to ignore this one?

File: clientapi/auth/login_jwt.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+	if !token.Valid {
+		return nil, nil, &util.JSONResponse{
+			Code: http.StatusForbidden,
+			JSON: jsonerror.Forbidden("Invalid JWT"),
+		}
+	}
+
+	r.Login.Identifier.User = c.Subject
+	r.Login.Identifier.Type = "m.id.user"
```
Comment (Danieloni1) [inline clientapi/auth/login_jwt.go:69]: linter says `string "m.id.user" has 3 occurrences, make it a constant` 🤷

File: clientapi/auth/password.go [hunk 3]
```diff
@@ -61,7 +65,35 @@ func (t *LoginTypePassword) LoginFromJSON(ctx context.Context, reqBytes []byte)
...
 	r := req.(*PasswordRequest)
-	username := strings.ToLower(r.Username())
+	if r.Identifier.Address != "" {
+		r.Address = r.Identifier.Address
+	}
+	if r.Identifier.Medium != "" {
+		r.Medium = r.Identifier.Medium
+	}
+	var username string
+	if r.Medium == "email" && r.Address != "" {
```
Comment (Danieloni1) [inline clientapi/auth/password.go:75]: Linter says `string "email" has 3 occurrences, make it a constant` 🤷

File: clientapi/auth/password.go [hunk 4]
```diff
@@ -99,11 +139,19 @@ func (t *LoginTypePassword) Login(ctx context.Context, req interface{}) (*Login,
...
+			t.rt.Act(localpart)
 			return nil, &util.JSONResponse{
 				Code: http.StatusForbidden,
-				JSON: jsonerror.Forbidden("The username or password was incorrect or the account does not exist."),
+				JSON: jsonerror.Forbidden("Invalid username or password"),
 			}
 		}
 	}
-	return &r.Login, nil
+	return &Login{
```
Comment (Danieloni1) [inline clientapi/auth/password.go:149]: I suppose automations pass with `password` login, right? 😬

File: clientapi/ratelimit/rt_failed_login.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	return true, 0
+}
+
+func (r *rateLimit) act() {
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
+	now := time.Now()
+	l := r.times.Len()
+	log.Print("len act: ", l)
```
Comment (Danieloni1) [inline clientapi/ratelimit/rt_failed_login.go:118]: Should we keep this log?

File: clientapi/ratelimit/rt_failed_login_test.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"testing"
+	"time"
+
+	"github.com/matryer/is"
+)
+
+func TestRtFailedLogin(t *testing.T) {
+	is := is.New(t)
+	dut := NewRtFailedLogin(&RtFailedLoginConfig{
```
Comment (Danieloni1) [inline clientapi/ratelimit/rt_failed_login_test.go:12]: Why did you choose the `dut` naming for this `RtFailedLogin` variable? 🤔

File: clientapi/ratelimit/rt_failed_login_test.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+	)
+	for i := 0; i < 3; i++ {
+		can, remaining = dut.CanAct("foo")
+		is.True(can)
+		is.Equal(remaining, time.Duration(0))
+		dut.Act("foo")
+	}
+	can, remaining = dut.CanAct("foo")
+	is.True(!can)
+	is.True(remaining > time.Millisecond*9)
```
Comment (Danieloni1) [inline clientapi/ratelimit/rt_failed_login_test.go:30]: Why specifically `time.Millisecond*9`? As far as I understood this should be dependent of `r.cfg.Interval`? Maybe I didn't get something? 🤔

File: internal/httputil/httpapi.go [hunk 8]
```diff
@@ -83,6 +83,57 @@ func MakeAuthAPI(
 	return MakeExternalAPI(metricsName, h)
 }
 
+// MakeContitionalAuthAPI turns a util.JSONRequestHandler function into an http.Handler which authenticates the request.
+// It passes nil device if header is not provided.
+func MakeContitionalAuthAPI(
```
Comment (Danieloni1) [inline internal/httputil/httpapi.go:88]: `MakeConditionalAuthAPI`* ?
Also in the comment :)

File: clientapi/routing/routing.go [hunk 9]
```diff
@@ -538,7 +540,7 @@ func Setup(
 	).Methods(http.MethodGet, http.MethodOptions)
 
 	v3mux.Handle("/account/password",
-		httputil.MakeAuthAPI("password", userAPI, func(req *http.Request, device *userapi.Device) util.JSONResponse {
+		httputil.MakeContitionalAuthAPI("password", userAPI, func(req *http.Request, device *userapi.Device) util.JSONResponse {
```
Comment (Danieloni1) [inline clientapi/routing/routing.go:543]: MakeConditionalAuthAPI

**Review by Danieloni1 — COMMENTED** [review]
That's a big one!
It all looks very good except for minor typos and other stuff I commented about.
Did all Dendrite tests pass?
Adminas automations pass for password, threePid, right?

File: clientapi/ratelimit/rt_failed_login_test.go (hunk 7 above)
Comment (PiotrKozimor) [reply clientapi/ratelimit/rt_failed_login_test.go:30]: We expect `remaining` to be slightly less than 10 ms, which is interval configured for tests. Hence 9 ms, but e.g. 9990 microseconds should work as well.

File: clientapi/ratelimit/rt_failed_login_test.go (hunk 6 above)
Comment (PiotrKozimor) [reply clientapi/ratelimit/rt_failed_login_test.go:12]: I realized it does makes sense in this context: https://en.wikipedia.org/wiki/Device_under_test, I will change it to `rtfl`

File: appservice/workers/transaction_scheduler.go [hunk 10]
```diff
@@ -58,38 +58,40 @@ func SetupTransactionWorkers(
...
+	id := 0
 	// Loop forever and keep waiting for more events to send
 	for {
 		// Wait for more events if we've sent all the events in the database
-		ws.WaitForNewEvents()
+		ws.WaitForNewEvents(id)
 
 		// Batch events up into a transaction
-		transactionJSON, txnID, maxEventID, eventsRemaining, err := createTransaction(ctx, db, ws.AppService.ID)
+		transactionJSON, txnID, maxEventID, _, err := createTransaction(ctx, db, ws.AppService.ID)
```
[the remark this answers is not in the document]
Comment (PiotrKozimor) [reply appservice/workers/transaction_scheduler.go:83]: I will remove it

**Comment by PiotrKozimor** [comment]
The linter error are now fixed. The `Guest users can join guest_access rooms` fails across all sytest versions, and it is somehow caused by our change in appservice component (it no longer fails when after `git restore --source=main appservice/`), which is veeery strange. Adminas automation pass nicely :) We will need to focus together on this failing test case.

File: clientapi/routing/routing.go (hunk 9 above)
Comment (PiotrKozimor) [reply clientapi/routing/routing.go]: @Danieloni1 comment refers to outdated code, I have changed it

**Review by Danieloni1 — APPROVED** [review]

**Pull request merged**